runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8570,23 +8567,6 @@ class INTERPRETER_CLASS
 				goto retry;
 			}
 		}
-#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
```

Comment:
See [here](https://bugs.openjdk.org/browse/JDK-8317766):

```
The `checkcast` instruction may opt to refer to a `CheckedType` constant instead of a `Class` constant. 

At run time, the `CheckedType` constant is first resolved (even if a `null` is on the stack). Then the checked type is enforced, possibly failing with a `ClassCastException` or a `NullPointerException`. 
```

Instead of removing this, it is better put a comment here. When checkedType is added/supported, we should throw an exception when casting null to a null-restricted checkedType.

Reply:
I added this back. Do we have an issue for CheckedType I can reference?

Reply:
We do not merge commented code in OpenJ9. Maybe I should clarify, you can remove the following block and add a comment on the null-restricted checkedType. 

> Do we have an issue for CheckedType I can reference in the comment?

You can create a new issue or use https://github.com/eclipse-openj9/openj9/issues/19460 (NullRestrictedArray there requires null restricted type support)

Reply:
Okay. I created this new issue and linked back to this discussion for the future. https://github.com/eclipse-openj9/openj9/issues/19764

Reply:
> add a comment on the null-restricted checkedType.
Not sure where you mean for this.